src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventRecordConstructor.java

```diff
@@ -163,6 +177,47 @@ private GenericRecord getPostHookEvent(HookContext hookContext, EventStatus stat
     return recordBuilder.build();
   }
 
+  private void retrieveApplicationResourceUsageReport(GenericRecordBuilder recordBuilder, ApplicationResourceUsageReport applicationResourceUsageReport) {
+    recordBuilder.set("YarnReportUsedResources", String.valueOf(applicationResourceUsageReport.getUsedResources().toString()));
+    callMethodWithReflection("getMemorySize", "YarnReportUsedResourcesMemory", recordBuilder, applicationResourceUsageReport.getUsedResources());
+    callMethodWithReflection("getVirtualCores", "YarnReportUsedResourcesVcore", recordBuilder, applicationResourceUsageReport.getUsedResources());
+
+    recordBuilder.set("YarnReportReservedResources", String.valueOf(applicationResourceUsageReport.getReservedResources().toString()));
+    callMethodWithReflection("getMemorySize", "YarnReportReservedResourcesMemory", recordBuilder, applicationResourceUsageReport.getReservedResources());
+    callMethodWithReflection("getVirtualCores", "YarnReportReservedResourcesVcore", recordBuilder, applicationResourceUsageReport.getReservedResources());
+
+    recordBuilder.set("YarnReportNeededResources", String.valueOf(applicationResourceUsageReport.getNeededResources().toString()));
+    callMethodWithReflection("getMemorySize", "YarnReportNeededResourcesMemory", recordBuilder, applicationResourceUsageReport.getNeededResources());
+    callMethodWithReflection("getVirtualCores", "YarnReportNeededResourcesVcore", recordBuilder, applicationResourceUsageReport.getNeededResources());
+
+    callMethodWithReflection("getNumUsedContainers", "YarnReportNumUsedContainers", recordBuilder, applicationResourceUsageReport);
```

Comment:
Done, I used map methodName -> fieldName and only for applicationResourceUsageReport, since for Resources we would need to also add object to that mapping too. In my understanding I can so it by adding Pair(which requires additional library import) or creating new class for holding those objects(,fieldName, className) which will complicate code instead of simplifying. 
Other option is to make map methodNameSuffix -> object, this will lead to generating method and field names programmatically, dont feel like its good approach.